### PR TITLE
notmuch: add page

### PR DESCRIPTION
**Pull request by Sadeeed**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

Ref: #1162

### CODE REVIEW

File: pages/common/notmuch.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+> A command-line based program for indexing, searching, reading, and tagging large collections of email messages.
+> More Information: <https://notmuchmail.org/manpages/>.
+
+- Configure for first use:
+
+`notmuch setup`
+
+- Add tags for all messages matching the search term:
+
+`notmuch tag +{{custom-tag}} {{"search-term"}}`
```
Comment (navarroaxel) [inline pages/common/notmuch.md:12]: We use snake_case for variable names and quotes outside:
```suggestion
`notmuch tag +{{custom_tag}} "{{search_term}}"`
```

File: pages/common/notmuch.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+- Remove tags for all messages matching the search term:
```
Comment (navarroaxel) [inline pages/common/notmuch.md:14]: is 1 tag?
```suggestion
- Remove a tag for all messages matching a search term:
```

File: pages/common/notmuch.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+`notmuch tag -{{custom-tag}} {{"search-term"}}`
```
Comment (navarroaxel) [inline pages/common/notmuch.md:16]:
```suggestion
`notmuch tag -{{custom_tag}} "{{search_term}}"`
```

File: pages/common/notmuch.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+`notmuch count --output={{messages|threads}} {{"search-term"}}`
+
+- Search for messages matching the given search term:
+
+`notmuch search --format={{json|text}} --output={{summary|threads|messages|files|tags}} {{"search-term"}}`
+
+- Limit the number of saerch results to X:
+
+`notmuch search --format={{json|text}} --output={summary|threads|messages|files|tags} --limit={{X}} {{"search-term"}}`
```
Comment (navarroaxel) [inline pages/common/notmuch.md:28]:
```suggestion
`notmuch search --format={{json|text}} --output={{summary|threads|messages|files|tags}} --limit={{X}} "{{search_term}}"`
```

File: pages/common/notmuch.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+- Create a reply template for a set of messages:
+
+`notmuch reply --format={{default|headers-only}} --reply-to={{sender|all}} {{"search-term"}}`
```
Comment (navarroaxel) [inline pages/common/notmuch.md:32]:
```suggestion
`notmuch reply --format={{default|headers-only}} --reply-to={{sender|all}} "{{search_term}}"`
```

File: pages/common/notmuch.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+`notmuch tag -{{custom-tag}} {{"search-term"}}`
+
+- Count messages matching the given search term:
+
+`notmuch count --output={{messages|threads}} {{"search-term"}}`
+
+- Search for messages matching the given search term:
+
+`notmuch search --format={{json|text}} --output={{summary|threads|messages|files|tags}} {{"search-term"}}`
```
Comment (navarroaxel) [inline pages/common/notmuch.md:24]:
```suggestion
`notmuch search --format={{json|text}} --output={{summary|threads|messages|files|tags}} "{{search_term}}"`
```

File: pages/common/notmuch.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+`notmuch tag +{{custom-tag}} {{"search-term"}}`
+
+- Remove tags for all messages matching the search term:
+
+`notmuch tag -{{custom-tag}} {{"search-term"}}`
+
+- Count messages matching the given search term:
+
+`notmuch count --output={{messages|threads}} {{"search-term"}}`
```
Comment (navarroaxel) [inline pages/common/notmuch.md:20]:
```suggestion
`notmuch count --output={{messages|threads}} "{{search_term}}"`
```

**Review by navarroaxel — CHANGES_REQUESTED** [review]
Hey @Sadeeed, I left some comments.

File: pages/common/notmuch.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
+# notmuch
+
+> A command-line based program for indexing, searching, reading, and tagging large collections of email messages.
+> More Information: <https://notmuchmail.org/manpages/>.
+
+- Configure for first use:
+
+`notmuch setup`
+
+- Add tags for all messages matching the search term:
```
Comment (navarroaxel) [inline pages/common/notmuch.md:10]:
```suggestion
- Add a tag for all messages matching a search term:
```

**Review by navarroaxel — CHANGES_REQUESTED** [review]

**Review by navarroaxel — APPROVED** [review]

**Review by sbrl — APPROVED** [review]
Thanks, @Sadeeed! Looks ok to me. Just made a teensy grammar fix.

**Comment by sbrl** [comment]
Just 1 command left in #1162! :tada:

**Pull request merged**